src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,46 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int index = random.nextInt(FigureList.values().length);
+        String figureName = FigureList.values()[index].name();
```

Comment:
Instead of calling `FigureList.values()` multiple times, it's better to store the result in a variable and use it to avoid multiple array creations.

Reply:
Agree with this comment